catalogd/internal/source/containers_image.go

```diff
@@ -431,3 +432,13 @@ func wrapTerminal(err error, isTerminal bool) error {
 	}
 	return reconcile.TerminalError(err)
 }
+
+func isImageUnpacked(unpackPath string) (bool, time.Time) {
+	if unpackStat, err := os.Stat(unpackPath); err == nil {
+		if !unpackStat.IsDir() {
+			panic(fmt.Sprintf("unexpected file at unpack path %q: expected a directory", unpackPath))
```

Comment:
I know this is a slight tangent from this PR, but I wonder if it would be better for us to just delete the file and unpack over top of where it was.